src/frequenz/sdk/data_ingestion/resampling/timeseries_resampler.py

```diff
@@ -0,0 +1,118 @@
+    ) -> None:
+        """Initialize the TimeSeriesResampler.
+
+        Args:
+            period_s: resampling frequency, in seconds
+            max_data_age_in_periods: max age that samples shouldn't exceed in order
+                to be used in the resampling function
+            resampling_function: function to be applied to a sequence of samples within
+                a resampling period to produce a single output sample
+        """
+        self._period_s = period_s
+        self._max_data_age_in_periods: float = max_data_age_in_periods
+        self._resampling_function: ResamplingFunction = resampling_function
+        self._resamplers: Dict[TimeSeriesId, IndividualTimeSeriesResampler] = {}
+
+    def add_time_series(self, time_series_id: TimeSeriesId) -> None:
+        """Create a new resampler for a specific time series.
+
+        Args:
+            time_series_id: time series id
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id already exists
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(
+                f"Resampler for time series {time_series_id} already exists!"
+            )
+
+        self._resamplers[time_series_id] = IndividualTimeSeriesResampler(
+            buffer_size=DEFAULT_BUFFER_SIZE,
+            period_s=self._period_s,
+            max_data_age_in_periods=self._max_data_age_in_periods,
+            resampling_function=self._resampling_function,
+        )
+
+    def remove_timeseries(self, time_series_id: TimeSeriesId) -> None:
+        """Remove a resampler for a specific time series.
+
+        Args:
+            time_series_id: time series id, for which to remove the resampler
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id doesn't exist
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(f"No resampler for time series {time_series_id} found!")
+
+        del self._resamplers[time_series_id]
+
+    def add_sample(self, time_series_id: TimeSeriesId, sample: Sample) -> None:
+        """Add a sample for a specific time series.
+
+        Args:
+            time_series_id: time series id, for which to remove the resampler
+            sample: sample to be added
+
+        Raises:
+            ValueError: if resampler for the provided timer_series_id doesn't exist
+        """
+        resampler = self._resamplers.get(time_series_id)
+        if resampler is None:
+            raise ValueError(f"No resampler for time series {time_series_id} found!")
+
+        resampler.add_sample(sample)
+
+    def resample(self) -> Dict[TimeSeriesId, Optional[Sample]]:
+        """Resample samples for all time series.
+
+        Returns:
+            mapping of time series ids and their newly resampled samples
+        """
+        now = datetime.now(tz=pytz.UTC)
```

Comment:
It was used in `microgrid_data` and `power_distributor` so I thought it could be used here too. Do you want me to replace all occurrences of `pytz.UTC` with `datetime.timezone.utc` ?